### DISCUSSION TITLE
Admin server can not save changes, error: 413 (Payload Too Large) and 404 (Not Found)

### DISCUSSION THREAD

**Discussion by silinzhong** [GMOD/jbrowse-components#2703] [Q&A]
Hi

I don't know how to use the admin server to save the changes i made. I installed the jbrowse2 in /var/www/html/jbrowse2. i started the admin server using jbrowse admin-server --target=config.json

If i login using the ip address followed by the port like 137.189.43.55:9090?adminKey=xxxxx and make some changes, i will get the 413 (Payload Too Large) err at the bottom on the track screen, as well as an error on the terminal.

If i login using the ip address without using the port 9090 like 137.189.43.55/jbrowse2?adminKey=xxxx and make changes, i will get the 404 (Not Found) err message.

I might have done something wrong with the add tracks as well. I put the bigwig files in the folder /var/www/html/SSD/... I then add the tracks using jbrowse add-track /SSD/Mp/xxx.bw -a ".." -n "..." --load inPlace --force. I can view the tracks when accessing the jbrowse using 137.189.43.55/jbrowse2. If i open the Jbrowse using 137.189.43.55:9090 (no matter with or w/o adminKey), it can not load the bigwig file as it is trying to find it at 137.189.43.55:9090/SSD/...

What should I do?

# error message displayed on laptop's chrome jbrowse2 screen 
Admin server error: 413 (Payload Too Large) <!DOCTYPE html> <html lang="en"> <head> <meta charset="utf-8"> <title>Error</title> </head> <body> <pre>PayloadTooLargeError: request entity too large<br> &nbsp; &nbsp;at readStream (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/raw-body/index.js:155:17)<br> &nbsp; &nbsp;at getRawBody (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/raw-body/index.js:108:12)<br> &nbsp; &nbsp;at read (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/body-parser/lib/read.js:77:3)<br> &nbsp; &nbsp;at jsonParser (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/body-parser/lib/types/json.js:135:5)<br> &nbsp; &nbsp;at Layer.handle [as handle_request] (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/express/lib/router/layer.js:95:5)<br> &nbsp; &nbsp;at trim_prefix (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/express/lib/router/index.js:323:13)<br> &nbsp; &nbsp;at /usr/local/lib/node_modules/@jbrowse/cli/node_modules/express/lib/router/index.js:284:7<br> &nbsp; &nbsp;at Function.process_params (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/express/lib/router/index.js:341:12)<br> &nbsp; &nbsp;at next (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/express/lib/router/index.js:275:10)<br> &nbsp; &nbsp;at cors (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/cors/lib/index.js:188:7)</pre> </body> </html>

# error message display on the web server terminal 
PayloadTooLargeError: request entity too large
    at readStream (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/raw-body/index.js:155:17)
    at getRawBody (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/raw-body/index.js:108:12)
    at read (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/body-parser/lib/read.js:77:3)
    at jsonParser (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/body-parser/lib/types/json.js:135:5)
    at Layer.handle [as handle_request] (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/express/lib/router/index.js:323:13)
    at /usr/local/lib/node_modules/@jbrowse/cli/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/express/lib/router/index.js:341:12)
    at next (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/express/lib/router/index.js:275:10)
    at cors (/usr/local/lib/node_modules/@jbrowse/cli/node_modules/cors/lib/index.js:188:7)

# error message when login using 137.189.43.55/jbrowse2/
Admin server error: 404 (Not Found) <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN"> <html><head> <title>404 Not Found</title> </head><body> <h1>Not Found</h1> <p>The requested URL was not found on this server.</p> <hr> <address>Apache/2.4.41 (Ubuntu) Server at 137.189.43.55 Port 80</address> </body></html>

**Comment by garrettjstevens** [comment]
I think you are doing the right thing when you get the 413, the problem is on our end. We need to increase the payload size of the `admin-server`.

As for not being able to view the tracks while accessing the admin server port, it looks like you're running `admin-server` from the `/var/www/html/jbrowse2` directory. Does it work if you instead run that from the `/var/www/html` directory (e.g. `jbrowse admin-server --target=jbrowse2/config.json`) and then access `137.189.43.55:9090/jbrowse2`?

**Reply by garrettjstevens** [reply]
FYI I've put in a PR to increase the payload size here: https://github.com/GMOD/jbrowse-components/pull/2705

**Reply by silinzhong** [reply]
thanks! I tested with another jbrowse installed with only 1 track and a small 3k config.json file. the admin server can now save the changes. Looking forward to the update to fix the max size!

Running the admin server from the /var/www/html directory as suggested indeed solved the track viewing problem in admin mode.

**Reply by cmdcolin** [reply]
New 1.6.5 release has this fix!

**Comment by silinzhong** [comment]
Dear Colin

Thanks for the fix!

I first run jbrowse upgrade and check the website’s help. It shows the version is 1.6.5 now. But when I use the admin-server, it still shows the same Payload too large error.

I also try to specify the body size limit but it doesn’t seem to be working.

***@***.***:/var/www/html$ jbrowse admin-server --target ./jbrowse2/config.json --bodySizeLimit 20mb
 ›   Error: Unexpected arguments: --bodySizeLimit, 20mb
 ›   See more help with --help

I also tried to reboot my webserver. Still not working. Is there something I need to do after upgrade to get the admin-server to work?

sorry for the trouble.
silin

**Reply by cmdcolin** [reply]
Hi @silinzhong you will also want to upgrade the @jbrowse/cli tool. Try running `npm install -g @jbrowse/cli` again to get the latest version of the CLI tools (the `jbrowse upgrade` command just gets the latest version of the website code)
